domains/certificates/Query.API/API.AppTests/ContractTests.cs

```diff
@@ -5,22 +5,27 @@
 using API.AppTests.Helpers;
 using API.AppTests.Infrastructure;
 using API.AppTests.Mocks;
+using API.IntegrationTest.Infrastructure;
 using FluentAssertions;
 using Xunit;
 
 namespace API.AppTests;
 
-public sealed class ContractTests : IClassFixture<QueryApiWebApplicationFactory>, IClassFixture<MartenDbContainer>, IDisposable
+public sealed class ContractTests : IClassFixture<QueryApiWebApplicationFactory>, IClassFixture<MartenDbContainer>, IClassFixture<RabbitMqContainer>, IDisposable
 {
     private readonly QueryApiWebApplicationFactory factory;
     private readonly DataSyncWireMock dataSyncWireMock;
 
-    public ContractTests(QueryApiWebApplicationFactory factory, MartenDbContainer marten)
+    public ContractTests(QueryApiWebApplicationFactory factory, MartenDbContainer marten, RabbitMqContainer rabbitMqContainer)
     {
         dataSyncWireMock = new DataSyncWireMock(port: 9001);
         this.factory = factory;
         this.factory.MartenConnectionString = marten.ConnectionString;
         this.factory.DataSyncUrl = dataSyncWireMock.Url;
+        this.factory.RabbitMqHost = rabbitMqContainer.Hostname;
+        this.factory.RabbitMqPort = rabbitMqContainer.Port.ToString();
+        this.factory.RabbitMqUsername = rabbitMqContainer.Username;
+        this.factory.RabbitMqPassword = rabbitMqContainer.Password;
```

Comment:
Suggestion: Instead of having 4 properties to set on factory we can add a method that takes all 4 arguments. This ensures that someone does not forget to set 1 of the 4 properties.

Reply:
If it is a method on the factory, then that method would have to store the arguments in backing fields, which is then used in the factory's ConfigureTestServices.

A maybe better approach is to expose a single property on the factory of type RabbitMqOptions. This type has all the properties anyway.